<?xml version="1.0"?>

<launch>

	<arg name="start_manager" default="true"/>
	<arg name="nodelet_manager" default="point_map_manager"/>
  <!-- start nodelet manager -->
  <node if="$(arg start_manager)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>
  <arg name="point_cloud" default="/registration/pc_aligned"/>

  <!-- node pkg="cob_3d_mapping_common" type="field_of_view_node" name="field_of_view_node" output="screen" /-->

	<!-- Start point map -->
  <node pkg="nodelet" type="nodelet" name="point_map_nodelet" args="load cob_3d_mapping_point_map/PointMapNodelet $(arg nodelet_manager)" output="screen">
  	<!--remap from="point_cloud2" to="/cam3d/depth_registered/points"/-->
  	<!--remap from="point_cloud2" to="/cam3d/rgb/points"/-->
  	<remap from="point_cloud2" to="$(arg point_cloud)"/>
  	<!--remap from="camera_info" to="/camera/depth_registered/camera_info"/-->
  	<remap from="map" to="point_map/map"/>
  	<!-- remap from="point_cloud2_aligned" to="point_map/pc_aligned"/-->
  	<remap from="clear_map" to="point_map/clear_map"/>
  	<remap from="get_map" to="point_map/get_map"/>
  	<remap from="set_map" to="point_map/set_map"/>

  	<param name="file_path" type="string" value="/tmp"/>
  	<param name="save" type="bool" value="false"/>
  	<param name="voxel_leafsize" type="double" value="0.03" />
  </node>

  <!-- node pkg="nodelet" type="nodelet" name="voxel" args="standalone pcl/VoxelGrid " output="screen">
    <remap from="~input" to="/point_map/pc_aligned" />
    <remap from="~output" to="/point_map/pc_aligned_vox" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -100.0
      filter_limit_max: 100.0
      leaf_size: 0.03
    </rosparam>
  </node-->

  </launch>
